jsonschema-core.xml

```diff
@@ -2827,32 +2820,61 @@ https://example.com/schemas/common#/$defs/count/minimum
                     </t>
                 </section>
 
-                <section title="Error or Annotation">
+                <section title="Errors">
                     <t>
-                        The error or annotation that is produced by the validation.
+                        Any errors produced by the validation.  This property MUST NOT
+                        be included if the validation was successful.  The value
+                        for this property MUST be an object where the keys are the names of
+                        keywords and the values are the error message produced by the
+                        associated keyword.  If the subschema itself is producing the
+                        error, that error MUST be listed with an empty string key.
                     </t>
                     <t>
-                        For errors, the specific wording for the message is not defined by this
+                        The specific wording for the message is not defined by this
                         specification.  Implementations will need to provide this.
                     </t>
                     <t>
-                        For annotations, each keyword that produces an annotation specifies its
-                        format.  By default, it is the keyword's value.
+                        The JSON key for this information is "errors".
+                    </t>
+                </section>
+
+                <section title="Annotations">
+                    <t>
+                        Any annotations produced by the validation.  This property MUST NOT
+                        be included if the validation was unsuccessful.  The value
+                        for this property MUST be an object where the keys are the names of
+                        keywords and the values are the annotations produced by the
+                        associated keyword.
                     </t>
                     <t>
-                        The JSON key for failed validations is "error"; for successful validations
-                        it is "annotation".
+                        Each keyword defines its own annotation data type (e.g. "properties"
+                        produces a list of keywords, whereas "$title" produces a string).
+                    </t>
+                    <t>
+                        The JSON key for this information is "annotations".
                     </t>
                 </section>
 
                 <section title="Nested Results">
                     <t>
-                        For "basic", this property will appear only at the root node and will hold
-                        all errors or annotations in a list.
+                        Nested results are generated from applicator keywords: keywords with
+                        subschemas or collections of subschemas as values.  Keywords which
```

Comment:
I'd suggest changing "keywords with subschema or..." to "keywords that create dynamic scopes by applying schema(s) to the instance" (technically they're not always lexically subschemas).